src/cpp/core/system/PosixCrypto.cpp

```diff
@@ -271,14 +274,32 @@ core::Error rsaInit()
 
    RAND_seed(entropy, ENTROPY_BYTES);
 
-   s_pRSA = ::RSA_generate_key(KEY_SIZE, 0x10001, NULL, NULL);
-   if (!s_pRSA)
-      return lastCryptoError(ERROR_LOCATION);
+   #if OPENSSL_VERSION_NUMBER < 0x10100000L
+      s_pRSA = ::RSA_generate_key(KEY_SIZE, 0x10001, NULL, NULL);
+      if (!s_pRSA)
+         return lastCryptoError(ERROR_LOCATION);
+
+      bn_n = s_pRSA->n;
+      bn_e = s_pRSA->e;
+   #else
+      BIGNUM *bn = BN_new();
+      BN_set_word(bn, RSA_F4);
+ 
+      s_pRSA = RSA_new();
+      int rc = ::RSA_generate_key_ex(s_pRSA, KEY_SIZE, bn, NULL);
+      BN_clear_free(bn);
+      if (rc != 1) {
+        return lastCryptoError(ERROR_LOCATION);
+        RSA_free(s_pRSA);
```

Comment:
This code won't be reached, given the prior `return` -- should this be moved up a line?